cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1057,11 +1063,47 @@ if (ARROW_WITH_ZSTD)
   endif()
 endif()
 
+# ----------------------------------------------------------------------
+# RE2 (required for Gandiva)
+if (ARROW_GANDIVA)
+  # re2
+  if ("${RE2_HOME}" STREQUAL "")
```

Comment:
This also needs to be added to the `$ARROW_BUILD_TOOLCHAIN` part at the top of the file. I will add and push here

Reply:
This was already there. This is failing for me for another reason so I'll push a fix